components/CMakeLists.txt

```diff
@@ -9,25 +9,17 @@
 #
 #===============================================================================
 
-# bmpersch
-# - Read modern cmake docs, use modern features
-# - Use find_package for trilinos and other TPLS
-
 cmake_minimum_required(VERSION 3.9)
 cmake_policy(SET CMP0057 NEW)
 set(CMAKE_CXX_STANDARD 17)
 
-# Store caseroot in the cache, so that, if cmake re-runs,
```

Comment:
CASEROOT is already always passed to cmake via `-DCASEROOT=xxx` so is already guaranteed to be in the cache.

Reply:
I don't really recall. Maybe I wanted to have a doc string in CMakeCache.txt (which you don't get with -Dblah only)? Long stretch though...

Reply:
But while we're on it, I think it's good practice to add
```
set (XYZ val CACHE TYPE doc)
```
for the CACHE vars that e3sm expects to receive. It is a way for our cmake files to self-document themselves. Otherwise one has to know how CIME will invoke cmake to figure out what comes from where...

Reply:
@bartgol , thanks. I will add that to the list of objectives for the full refactor.